streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/HighAvailabilityTaskAssignor.java

```diff
@@ -124,11 +132,20 @@ private static void assignActiveStatefulTasks(final SortedMap<UUID, ClientState>
             ClientState::assignActive,
             (source, destination) -> true
         );
+
+        if (rackAwareTaskAssignor != null && rackAwareTaskAssignor.canEnableRackAwareAssignor()) {
+            final int trafficCost = configs.rackAwareAssignmentTrafficCost == null ?
```

Comment:
Why do we get `trafficCost` each time? Seems we should set it up in the constructor just a single time (or implement `Configurable` interface)? (Same for `nonOverlapCost`)

Reply:
Because the costs are different for stateful and stateless tasks. So setting in constructor means we need to pass in two RackAwareTaskAssignor objects.